tests/test_data_handling.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+import requests
+import json
+
+from fobi_custom.plugins.form_handlers.collect_data.fobi_form_handlers import CollectDataPlugin
+from fobi.dynamic import assemble_form_class
+
+from pldp.models import Survey, SurveyRow, SurveyComponent
+
+
+@pytest.mark.django_db
+def test_data_handler(form_entry, location, study, form_element, mocker):
+    component_data = json.loads(form_element.plugin_data)
+    saved_data = 5
+
+    request = mocker.MagicMock(spec=requests.Response)
+    form_class = assemble_form_class(form_entry=form_entry)
+
+    # Instantiate an instance of the `assemble_form_class`.
+    form_class = assemble_form_class(form_entry=form_entry)
+    form = form_class(data={component_data['name']: saved_data})
+
+    assert form.is_valid()
+
+    # A valid CollectDataPlugin requires a DynamicForm, a FormEntry, and a post request
+    plugin = CollectDataPlugin()
+    plugin.run(form=form,
+               form_entry=form_entry,
+               request=request)
+
+    # Check that a survey has been created with a linked row and component
+    survey = Survey.objects.first()
+    rows = SurveyRow.objects.filter(survey=survey)
+
+    # Grab the row object itself out of the filtered row queryset, in order to
+    # find its linked components
+    row = rows[0]
+    components = SurveyComponent.objects.filter(row=row)
+    component = components[0]
+
+    # Check that the handler has created a single row and linked component
+    assert len(rows) == 1
+    assert len(components) == 1
+
+    # Check survey has the correct values
```

Comment:
Most of this code does not need the sort of obvious commenting you have here. 

If you feel like the code needs some explanation, you should place that in a docstring at the top of the text, i.e., right after line 12. It's not a requirement of this PR. Just something to think about.